core/src/entry.rs

```diff
@@ -160,13 +161,16 @@ impl Entry {
 
 pub fn hash_transactions(transactions: &[Transaction]) -> Hash {
     // a hash of a slice of transactions only needs to hash the signatures
-    let mut hasher = Hasher::default();
-    transactions.iter().for_each(|tx| {
-        if !tx.signatures.is_empty() {
-            hasher.hash(&tx.signatures[0].as_ref());
-        }
-    });
-    hasher.result()
+    let signatures: Vec<_> = transactions
+        .iter()
+        .map(|tx| tx.signatures[0].as_ref())
```

Comment:
should we be hashing every signature in a transaction or just one of 'em?

Reply:
My gut says all of them, but the legacy code only did the first.  I meant to dig through blame to see if there were ReasonsTM

Reply:
Looks like it may have been done in the blind when adding support for multiple signatures.

https://github.com/solana-labs/solana/commit/cda9ad8565eb99d4aac2bb9d46a986fcf056f479#diff-d0e9b1d1df1c5795eac22a324e40477eL195

Reply:
prolly ok to hash just one, as any (valid) one will cover all the TX data

Reply:
Should probably cover them all in the case of a mutlisig though, no?

Also won't be great to handle legacy entries if we decide in the future that we should've been hashing them all.

Reply:
yes, I think you've convinced me

Reply:
Should extending the signature hashing be done here, now or just add an issue and look into it later?  I'm not sure if there would be any side effects, etc.

Reply:
should be no side effects.  go for it.

Reply:
https://github.com/solana-labs/solana/pull/4644/commits/905c261afad26293110f16f5aa206b312dff2dd0

It doesn't seem like any of the entry tests include multiple signatures and it isn't clear what the quickest way to add one would be.  Is it enough to add multiple instructions with different signers to the same TX?